### PR TITLE
`aud` claim support

### PR DESCRIPTION
**Pull request by sashaCher**
### What does this PR do?

Enables `authn-jwt` to support `aud` claim according [RFC](https://datatracker.ietf.org/doc/html/rfc7519).

#### Change log
- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [ ] The changes in this PR do not affect the Conjur API

### CODE REVIEW

**Review by nessiLahav — CHANGES_REQUESTED** [review]
Just add limitation that audience variable supports only singe value and not array

File: app/domain/authentication/authn_jwt/validate_and_decode/fetch_audience_value.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          resource_class: ::Resource,
+          fetch_required_secrets: ::Conjur::FetchRequiredSecrets.new,
+          logger: Rails.logger
+        },
+        inputs: %i[authentication_parameters]
+      ) do
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingAudienceValue.new)
+          
+          return empty_audience_value unless audience_resource_exists?
```
Comment (nessiLahav) [inline app/domain/authentication/authn_jwt/validate_and_decode/fetch_audience_value.rb:18]: Why? it should be similar to `FetchIssuerValue` command class

**Comment by nessiLahav** [comment]
@sashaCher We need to decide what state if claim is mandatory to validate, the current behavior is:
iss,iat,nbf = if they exist in the token, we must to validate them

In this design you suggest otherwise:
If aud  exists in token is not must to validate
But if audience is configured it means it is must to be validated 

I think aud should act the same as other claims. (we can add in the future configuration of mandatory standard claim to validate)

@benoaviram @hilagross WDYT?

**Comment by sashaCher** [comment]
@nessiLahav I've added limitation and emphasised that `aud` is mandatory once variable exists

**Comment by sashaCher** [comment]
@nessiLahav 
1. It's a shame that we have no mechanism that allows customer to define claims mentioned by you as mandatory.
2. In any case we have to respect them if they are present
3. The behaviour should be consistent:
**Given:** As a authenticator owner I've defined the `audience` variable and provided it with valid value
**When** The authenticator receives a token without `aud` claim at all
**Then** The authentication failed

**Comment by nessiLahav** [comment]
> @nessiLahav
> 
> 1. It's a shame that we have no mechanism that allows customer to define claims mentioned by you as mandatory.
> 2. In any case we have to respect them if they are present
> 3. The behaviour should be consistent:
>    **Given:** As a authenticator owner I've defined the `audience` variable and provided it with valid value
>    **When** The authenticator receives a token without `aud` claim at all
>    **Then** The authentication failed

I am OK with the decision that we are enforcing the audience by the policy but as you mention in bullet 1, i think the UX is problematic/confusing:

**exp** - is mandatory hardcoded 
**iss** - is mandatory if exists in token while `issuer` variable is optional 
**iat, nbf** - are mandatory if exist in token
**aud** - is mandatory if `audience` variable is configured
**jti** - is ignored

**Comment by hilagross** [comment]
@nessiLahav @sashaCher 
if `audience` variable exists then it should must be in token and should be validated.
the variable itself should be optional, same as `iss`

I'm trying to understand if it's possible to verify `aud` claim that is in the token (same as other claims behaviour)

**Review by nessiLahav — APPROVED** [review]
LGTM

**Review by nessiLahav — APPROVED** [review]
LGTM

**Pull request merged**